python/cuml/ensemble/randomforest_common.pyx

```diff
@@ -241,13 +241,16 @@ class BaseRandomForestModel(Base):
             if y_dtype != np.int32:
                 raise TypeError("The labels `y` need to be of dtype"
                                 " `int32`")
-            self.classes_ = rmm_cupy_ary(cp.unique, y_m)
-            self.num_classes = len(self.classes_)
+            temp_classes = rmm_cupy_ary(cp.unique, y_m)
```

Comment:
```suggestion
            self._classes = CumlArray(cp.unique(y_m))
```
Doing this instead of a temp array might be slightly simpler, and I would suggest that this might be a good moment to change this to use the decorator https://github.com/rapidsai/cuml/blob/4a92adcab81811064d74e390b6a23d71b1ed8423/python/cuml/common/memory_utils.py#L38 intead of rmm_cupy_ary (makes the code nicer)

Reply:
I can change to the decorator (Have done this in a couple other places). Regarding the use of the temp array, I had to do this because the for loop with `CumlArray` doesnt work. So a couple of lines down, these statements failed:
```python
for i in range(self.num_classes):
   if i not in self._classes:
```
I figured since converting between `CumlArray` and `cupy.ndarray` is a short operation, its best to keep it a similar to how it was before, and then convert after the loop. The other option would require making the `in` statement work with `CumlArray` which seems like a different issue. What do you think?

Reply:
Resolved after discussing with @dantegd